src/librustc_mir/build/expr/into.rs

```diff
@@ -305,9 +303,12 @@ impl<'a,'tcx> Builder<'a,'tcx> {
                             -> BlockAnd<()>
         where F: FnOnce(&LoopScope) -> BasicBlock
     {
-        let loop_scope = self.find_loop_scope(span, label);
-        let exit_block = exit_selector(&loop_scope);
-        self.exit_scope(span, loop_scope.extent, block, exit_block);
+        let (exit_block, extent) = {
+            let loop_scope = self.find_loop_scope(span, label);
+            loop_scope.might_terminate = true;
```

Comment:
@nikomatsakis 

Something like

```
'a: while 1 != 2 {
     loop { continue 'a }
}
```

is a valid code and essentially terminates the loop inside of which the expression resides?